core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -465,13 +476,27 @@ public Team getPrimaryTeam(UUID uuid, boolean includeObservers) {
 
   private record AggStat<T extends Number & Comparable<T>>(
       StatType type, T value, Set<UUID> players) {
-    public AggStat<T> track(UUID uuid, StatHolder stat) {
-      T newVal = (T) stat.getStat(type);
+    public AggStat<T> track(UUID uuid, MatchPlayer player, StatHolder stat) {
+      T newVal = getStat(player, stat);
+      if (newVal == null) return this;
       int cmp = value.compareTo(newVal);
       if (cmp > 0) return this;
       if (cmp < 0) players.clear();
       players.add(uuid);
       return cmp == 0 ? this : new AggStat<>(type, newVal, players);
     }
+
+    private T getStat(MatchPlayer player, StatHolder stat) {
+      return (T) getStatValue(type, player, stat);
+    }
```

Comment:
```suggestion
    public AggStat<T> track(UUID uuid, MatchPlayer player, StatHolder stat) {
      T newVal = (T) getStatValue(type, player, stat);
      if (newVal == null) return this;
      int cmp = value.compareTo(newVal);
      if (cmp > 0) return this;
      if (cmp < 0) players.clear();
      players.add(uuid);
      return cmp == 0 ? this : new AggStat<>(type, newVal, players);
    }
```
no need for the extra method used exactly once

Reply:
Sure